electrode-ota-server-dao-mariadb/src/dao/PackageDAO.ts

```diff
@@ -63,6 +64,38 @@ export default class PackageDAO extends BaseDAO {
 
     public static async getNewestApplicablePackage(connection: IConnection, deploymentKey: string,
         tags: string[] | undefined, appVersion: string | undefined): Promise<PackageDTO | void> {
+        const deployment = await DeploymentDAO.deploymentForKey(connection, deploymentKey);
+        const deploymentId = deployment.id;
+
+        const isTagsRequested = tags && tags.length > 0;
+        const key = isTagsRequested ? "tags" : "notags";
+        const allQueries:{[key:string]: string;} = {
+            "tags": PackageQueries.getMostRecentPackageIdByDeploymentAndTags,
+            "notags": PackageQueries.getMostRecentPackageIdByDeployment
+        }
+        const allParams:{[key:string]:any} = {
+            "tags": [deploymentId, tags, deploymentId],
+            "notags": [deploymentId]
+        }
+        const packages = await PackageDAO.query(connection, allQueries[key], allParams[key]);
+        if (packages && packages.length > 0) {
+            if (appVersion) {
+                for (let i = 0; i < packages.length; i++) {
+                    if (semver.satisfies(appVersion, packages[i].appVersion)) {
+                        return await PackageDAO.packageById(connection, packages[i].package_id);
```

Comment:
your returning the first one that matches.  what if there are multiple packages that match?

Reply:
Always we match the latest (in terms of time) even thought if multiple packages are available.

Reply:
is there a test that tests it will return the latest?

Reply:
I'll check the tests, between check the previously used `getNewestApplicablePackageByNonSemver` method that matches for the exact `appVersion` in the query but returns only the one latest package from the result. It's just the same.

Reply:
check the test link for latest release event with couple of specific appVersion release
https://github.com/electrode-io/electrode-ota-server/blob/master/electrode-ota-server-dao-mariadb/test/ElectrodeOtaDaoRdms-spec.ts#L1514

Reply:
Added more tests for `getNewestApplicablePackage` to test all possible semver ranges: 
https://github.com/electrode-io/electrode-ota-server/pull/139/files#diff-793371ab800a9844ce9e83a3c592f284